phantazm-api/src/main/java/com/github/phantazmnetwork/api/config/VariantConfigProcessor.java

```diff
@@ -42,6 +42,10 @@ public VariantConfigProcessor(@NotNull Function<Key, ConfigProcessor<? extends T
             throw new ConfigProcessException("no subprocessor");
         }
 
+        if (element instanceof ConfigNode node) {
+            return processor.dataFromElement(node);
+        }
+
         return processor.dataFromElement(element.getElementOrThrow("data"));
```

Comment:
what if the subprocessor needs data in the root element? 

ex. 
```txt
PhantazmPlayer
- serialKey: phantazm:veryaverage
- type: PLAYER
- name: VeryAverage
- data: 
  - numberOfWeedsSmoked: 69420
```

AverageInfo needs the `numberOfWeedsSmoked` field to be constructed, in addition to the type and name; wouldn't it be better to pass on the entire cfgelement instead of just the data

Reply:
node is the root element? and wouldn't numberOfWeedsSmoked be at the same level as PhantazmPlayer?

Reply:
yes, node is the root element

> and wouldn't numberOfWeedsSmoked be at the same level as PhantazmPlayer
numberOfWeedsSmoked is specific to VA, not all phantazmplayers smoke

Reply:
> numberOfWeedsSmoked is specific to VA, not all phantazmplayers smoke
lol
anyway numberOfWeedsSmoked would just be grouped with all the other fields, the subprocessor will read the fields that it needs to function (which also includes type and name)

Reply:
actually i just realized something, the second return statement:
```java
return processor.dataFromElement(element.getElementOrThrow("data"));
```

will always fail because element cannot be a ConfigNode when it is reached

Reply:
is there anything wrong with letting numberOfWeedsSmoked be on the same level as everything else?

Reply:
there is not, but then there's no point to having the `data` field anymore

Reply:
the root node is always a ConfigNode (which makes sense because it has to contain `serialKey`) and you already return whatever element the subProcessor spits out, so i think it's okay to just throw an exception if `!element.isNode()`

Reply:
so you either have
```yml
PhantazmPlayer
- serialKey: phantazm:veryaverage
- weedsSmoked: 9001
```

in which case VeryAverageInfo is constructed from the root element (PhantazmPlayer) 

or you have
```java
PhantazmPlayer
- serialKey: phantazm:veryaverage
- serialData:
  - weedsSmoked: 9001
```

in which case VeryAverageInfo is only created from the serialData element

either of this approaches seems fine to me, but i don't understand why both are supported

Reply:
well if serialData exists it is *supposed* to be a non-ConfigNode - should I verify that? or merge serialData node with root node?